### ISSUE TITLE
Unexpected 416 response when no results to return and using limit& offset in combination with count=exact header

### ISSUE THREAD

**Issue by dmijatovic** [PostgREST/postgrest#3078]
### Environment

* PostgreSQL version: Docker image postgres:15.4 
* PostgREST version:  Docker image postgrest/postgrest:v11.2.1
* Operating system: Linux (mint/ubuntu)

### Description of issue

I have noticed this errors recently after we performed Postgres and PostgREST upgrade. 

When using limit&offset and count=exact in the header of the fetch request the api returns 416 response instead of 200 with empty array in case the table (in this example RPC) is empty (no records to return). The offset is correctly set to 0 (first page). 

#### Request (Javascript, fetch)

```Javascript
const url = http://localhost:3000/api/v1/rpc/citation_by_software?offset=0&limit=3

const resp = await fetch(url, {
  method: 'GET',
  headers: {
    'Prefer': 'count=exact'
  }
})
```

#### Response
The error message returned from postgrest states in the details `An offset of 0 was requested, but there are only 0 rows.` (see below). The error details seem to me incorrect.

```json
{
    "code": "PGRST103",
    "details": "An offset of 0 was requested, but there are only 0 rows.",
    "hint": null,
    "message": "Requested range not satisfiable"
}
```

![image](https://github.com/PostgREST/postgrest/assets/9204081/82ee47be-224a-4e5f-ae8c-24ae9e65b65b)

#### Expected behavior
**Return status 200 with empty array object when table/rpc/view/selection has no (valid) results, as it was the case in the previous versions of postgrest. When header `count=exact` is NOT used expected response is retured (status:200, data:[]) but when using count=exact header I receive described error 416 response**. Please note that in other request we also use various filtering and ordering params. I omitted these to make the base case simpler.

(Steps to reproduce: Include a minimal SQL definition plus how you make the request to PostgREST and the response body)

I think the problem description is clear and easy to reproduce on any empty table. However let me know if you need more information.

Thanks for all nice development work you do with postgrest :-)

**Comment by laurenceisla** [comment]
I think this was fixed in v11.2.2 (see https://github.com/PostgREST/postgrest/pull/3025)? Could you try with that version and see if that's the case?

**Comment by dmijatovic** [comment]
@laurenceisla Yes, I confirm that is fixed in v11.2.2
Thanks!
